bots: Rename "joined" to "created" in bot profile

vanshgit1111 opened the pull request:
Fixes #37251.

Bots are created, not joined. The previous "joined" label was confusing,
so this updates it to better match how bots are added.

Bot profile is changed, showing "created" only - 


**Screenshots and screen captures:<img width="592" height="620" alt="Screenshot 2025-12-30 at 11 00 52 PM" src="https://github.com/user-attachments/assets/0f6cea8d-a51b-4256-a27c-f57f9a2816a7" />**

Normal user profile is unchanged, showing "joined" only - 

<img width="597" height="617" alt="Screenshot 2025-12-31 at 3 01 26 AM" src="https://github.com/user-attachments/assets/7f9e6e62-5882-4766-b6f7-8f0f1a45edff" />

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->


- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks for the fix! Could you please update your commit message to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages)? 

You will also need to complete include a screenshot showing that a regular user's profile is not affected.

vanshgit1111 commented:
Updated the commit message to follow the commit style guidelines and added a screenshot
showing that a regular user's profile is unchanged.

<img width="1509" height="818" alt="Screenshot 2025-12-31 at 3 01 39 AM" src="https://github.com/user-attachments/assets/5caef457-06ae-4fd1-8405-e1f961664211" />

alya commented:
You can tag @sahil839 for review when ready.

@sahil839  I haven't tested manually.

sahil839 commented:
The commit message doesn't follow the guidelines, i.e. it does not mention the issue it fixes. Also would be good to add a line about why this change was needed like you added it in the PR description.

alya commented:
@vanshgit1111 Please add @MohanReddy369 as a co-author in your commit message if you used any of the work in #37259.

sahil839 commented:
@timabbott ready for your review.
